aisim/atoms.py

```diff
@@ -70,22 +73,33 @@ def time(self, value):
         self._time = value
 
     @property
-    def state_vectors(self):
+    def state_kets(self):
         """
-        optional n × m array, representing the internal state of the m level system of each atom
+        optional n × 1 x m array, representing the ket vectors of the m level system of each atom
         """
-        return self._state_vectors
+        return self._state_kets
 
-    @state_vectors.setter
-    def state_vectors(self, new_state_vectors):
-        new_state_vectors = np.array(new_state_vectors)
-        if new_state_vectors.ndim == 1:
+    @state_kets.setter
+    def state_kets(self, new_state_kets):
+        if isinstance(new_state_kets, list):
+            new_state_kets = np.array([new_state_kets]).T
+        else:
+            new_state_kets = new_state_kets
+        if len(new_state_kets) == 2:

```

Comment:
what was wrong with the old test, where the ndim attribute is checked? Now it only works for 2 level systems

Reply:
yes this was not good. changed it again to check for two dimensions (row and columns of state_vector).
complete sets of kets should have three dimensions